apps/utils/start_server.py

```diff
@@ -0,0 +1,121 @@
+)
+@click.option(
+    "--base-branch", type=str, default=None, help="Name of base branch (if not given, 'master' will be used)."
+)
+def cli(new_branch: Optional[str] = None, base_branch: Optional[str] = None) -> None:
+    if not GITHUB_TOKEN:
+        log.error("Define GITHUB_TOKEN in your ETL .env file. Then run this tool again.")
+        return
+
+    # Initialize a repos object at the root folder of the etl repos.
+    repo = Repo(BASE_DIR)
+
+    # List all local branches.
+    local_branches = [branch.name for branch in repo.branches]
+
+    # Update the list of remote branches in the local repository.
+    origin = repo.remote(name="origin")
+    origin.fetch()
+    # List all remote branches.
+    remote_branches = [ref.name.split("origin/")[-1] for ref in origin.refs if ref.remote_head != "HEAD"]
+
+    if base_branch is None:
+        # If a base branch is not specified, assume that it will be 'master'.
+        base_branch = "master"
+
+    # Ensure the base branch exists in remote (this should always be true for 'master').
+    if base_branch not in remote_branches:
+        log.error(
+            f"Base branch '{base_branch}' does not exist in remote. "
+            "Either push that branch (git push origin base-branch-name) or use 'master' as a base branch. "
+            "Then run this tool again."
+        )
+        return
+
+    if new_branch is None:
+        # If not explicitly given, the new branch will be the current branch.
+        new_branch = repo.active_branch.name
+        if new_branch == "master":
+            log.error("You're currently on 'master' branch. Use '--new-branch ...' to create a new branch.")
+            return
+    else:
+        # Ensure the new branch does not already exist locally.
+        if new_branch in local_branches:
+            log.error(
+                f"Branch '{new_branch}' already exists locally."
+                "Either choose a different name for the new branch to be created, "
+                "or switch to the new branch and run this tool without specifying a new branch."
+            )
+            return
+        log.info("Create new branch and switch to it.")
+        repo.git.checkout("-b", new_branch)
+
+    # Ensure the new branch does not already exist in remote.
+    if new_branch in remote_branches:
+        log.error(
+            f"New branch '{new_branch}' already exists in remote. "
+            "Either manually create a pull request from github, or use a different name for the new branch."
+        )
+        return
+
+    log.info("Creating an empty commit.")
+    repo.git.commit("--allow-empty", "-m", "Start a new staging server")
+
+    log.info("Pushing the new branch to remote.")
+    repo.git.push("origin", new_branch)
+
+    log.info("Creating a draft pull request.")
+    headers = {"Authorization": f"token {GITHUB_TOKEN}", "Accept": "application/vnd.github.v3+json"}
+    data = {
+        "title": f":construction: Draft PR for branch {new_branch}",
```

Comment:
I'd appreciate having an option to specify PR title and message of the first commit (they tend to be identical quite often). For instance
```
etl d start-server --new-branch my-branch -c ":bar_chart: My dataset update"
```

Reply:
Hmm, I thought it's a good idea to have a generic title for the draft PR. This way it's easier to spot if some Draft PR was forgotten.
Then, when making it a real PR, ready for review, the user will have to assign a proper title to it. What do you think? Assigning the title from the terminal is also a bit inconvenient. But if you think we should have the option, I'm happy to do that.

Reply:
Changing the title in PR is fine, but changing the commit message (to the same title) is annoying. Having an option to set both in CLI would be ultimate.

> This way it's easier to spot if some Draft PR was forgotten.

I don't think that's a big deal. We don't have that many PRs anyway.